image_tools/src/cam2image.cpp

```diff
@@ -101,7 +101,7 @@ int main(int argc, char * argv[])
 
   // Initialize a ROS 2 node to publish images read from the OpenCV interface to the camera.
   auto node = rclcpp::Node::make_shared("cam2image");
-  auto node_logger = node->get_logger();
+  rclcpp::Logger node_logger = node->get_logger();
```

Comment:
Out of curiosity, why do you think that?

Reply:
in http://ci.ros2.org/job/ci_windows/3833/ it failed at the decltype check in the macro. it's just a guess though :)

Reply:
I see, that might be, I didn't consider it was interacting with decltype, but even then it _should_ work :). Hopefully you'll find the issue.

Reply:
thanks to your pointers I opened https://github.com/ros2/rclcpp/pull/423 and the usage in the lambda below can pass the type check